dvc/exceptions.py

```diff
@@ -332,13 +332,12 @@ class CollectCacheError(DvcException):
     pass
 
 
-class RemoteNotSpecifiedInExternalRepoError(DvcException):
-    def __init__(self, url, cause=None):
-        super(RemoteNotSpecifiedInExternalRepoError, self).__init__(
+class NoRemoteInExternalRepoError(DvcException):
+    def __init__(self, url):
+        super(NoRemoteInExternalRepoError, self).__init__(
             "No DVC remote is specified in the target repository '{}'".format(
```

Comment:
Well, its not a line you're changing but the text could be slightly improved:

`"No DVC remote is specified in target repository '{}'."` (no "the" and added period)

That's my only suggestion.

Reply:
Also are we sure we're talking about a target "repository"? Or can it just be a "project".

Repository: basically referring to a Git repo.
Project: More ambiguous term but typically means a DVC project initialized with `dvc init` (whether or not it sits on top of Git).

Reply:
Project makes sense, however, if you will take a look at `exceptions.py` "dvc repository"  is usually used. To keep consistency I would leave `repository` and if we agree to change `repostiory` to `project` I would just prepare one change to fix all exceptions. How does it sound?

Reply:
Would also prefer "repository".

Reply:
OK. Fixed in #2881.